Added link to restoring backup page from the backup integration

dannytsang opened the pull request:
## Proposed change
Added image of the onboarding dialogue when restoring from a backup from a new installation.


## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #25050

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

dannytsang committed:
Added link to restoring backup page from the backup integration.

Added image of the onboarding diaglue when restoring from a backup from a new installation.

dannytsang committed:
Added line break between heading and body.

frenck commented on source/_integrations/backup.markdown:
```diff
@@ -66,3 +66,7 @@ automation:
+
+## Restoring from a backup
+
+To restore from a backup, see [Restoring a Backup on a new install](/common-tasks/supervised/#restoring-a-backup-on-a-new-install)
```
This is partially correct, but not was meant in the original issue/feedback raised.

The linked section is how to restore it onto an OS installation. It does not provide information on how to restore it onto a Container or Core installation type.

dannytsang replied:
@frenck I have added new pages for core and container backup restores. I used file paths used in the install guides so hopefully there is a bit of continuity.

I removed the addon folder from the directories backed up in core and container because I believe addons are only available for OS and supervised. I haven't verified this I went through the install to verify the folder paths but not to the point of running it. If you can confirm this then I hope all the other changes should address the feedback.

dannytsang committed:
Updated ways to restore from a backup when running HASS OS or Supervised.

dannytsang committed:
Refactored common sections.

Created and reused moving backups to a different location and creating backups from the UI. These sections are common across installation types.

dannytsang committed:
Created restore pages for core and container.

Updated backup integration with references to core and container installs.

frenck commented on source/_includes/common-tasks/backups_container.md:
```diff
@@ -0,0 +1,26 @@
+
+Start Home Assistant container.
+
+{% include common-tasks/backups_offsite.md %}
```
These do not apply? The ones listed in that snippet are add-ons, add-ons are not available on container installation types.

frenck commented on source/_includes/common-tasks/backups.md:
```diff
@@ -48,15 +45,4 @@ When the restore is complete, Home Assistant will restart to apply the new setti
-- [Remote Backup (scp/rsync/rclone)](https://github.com/ikifar2012/remote-backup-addon)
-- [Samba backup](https://github.com/thomasmauerer/hassio-addons/tree/master/samba-backup)
-- [Syncthing](https://github.com/Poeschl/Hassio-Addons/tree/main/syncthing)
+{% include common-tasks/backups_offsite.md %}
```
I don't think this has more use cases and therefore don't need to be extract into a snippet.

frenck commented on source/_includes/common-tasks/backups_create.md:
```diff
@@ -0,0 +1,8 @@
+### Making a Backup from the UI
```
This is not as common as it looks; creating backups is not the same for every installation method. The described steps below apply to installation types that ship with a Home Assistant Supervisor.

dannytsang committed:
Removed directories listed as per @frenck comment.

Changed it just contain the configuration directory.

dannytsang committed:
Removed offsite backup snipper because it just applies to supervised installs.
